Switch from Travis CI to CircleCI

gforsyth opened the pull request:
Jobs set to run in parallel and cache their respective conda environments.

Resolves #120 

Someone with admin rights will have to enable to circle hooks, and then I can rebase to trigger the job to make sure everything is working.

hameerabbasi commented on .circleci/config.yml:
```diff
@@ -0,0 +1,103 @@
+          command: |
+            /home/circleci/miniconda/envs/py36-sparse-test/bin/codecov
+
+  build_27:
```
There seems to be a lot of repetition between the two builds. Is there some way to bring that down?

hameerabbasi commented on ci/environment-2.7.yml:
```diff
@@ -0,0 +1,14 @@
+  - nomkl
+  - numba
+  - pip:
+      - pytest-flake8
```
Can we move this to conda-forge?

hameerabbasi commented on ci/environment-3.6.yml:
```diff
@@ -0,0 +1,14 @@
+  - nomkl
+  - numba
+  - pip:
+      - pytest-flake8
```
Same as above.

gforsyth commented:
Added a badge to the README and switched the environment files to use conda-forge instead of defaults.  There might be some way to avoid the build repetition, but I'm not familiar enough with circle 2 to know where (and to not bork up the cache restore).

hameerabbasi commented:
Hmm. Could you make a minor change? CircleCI was set to not build forked PRs.

hameerabbasi commented:
I turned off Travis builds completely, and made CircleCI required instead. Can you re-trigger with some change?

hameerabbasi commented on .circleci/config.yml:
```diff
@@ -0,0 +1,103 @@
+            pip install -e .[tests]
+      - run:
+          command: |
+            /home/circleci/miniconda/envs/py36-sparse-test/bin/pytest --flake8
```
Can you cd into the project directory and run `pytest`? It'll automatically pick up `pytest.ini` that way. You also won't need `--flake8`, it's already in `pytest.ini`.

This also makes the coverage not generate.

gforsyth commented:
Ok, should be all set now.  Had to properly set path for the `codecov` call because apparently `codecov` is calling the `coverage` entrypoint with `subprocess` (or something similarly strange)

hameerabbasi commented:
Thanks, @gforsyth! This is in.
